BookLibrary/src/main/java/Library.java

```diff
@@ -0,0 +1,96 @@
+import java.util.*;
+
+public class Library {
+    private List<Book> books;
+    private Set<Reader> readers;
+    private Map<Reader,List<Book>> registeredReaders;
+
+    public Library(){
+        this.books = new ArrayList<>();
+        this.readers = new HashSet<>();
+        this.registeredReaders = new HashMap<>();
+    }
+
+    public void addBook(String name, String author){
+//        if (this.books.isEmpty()){
+//            this.books.add(new Book(1, name, author));
+//        } else{
+//            this.books.add(new Book(this.books.size()+1, name, author));
+//        }
+        this.books.add(new Book(this.books.size()+1, name, author));
+    }
+    public Book getBookByID(int bookId){
+        Book book = null;
+        for(Book b : books){
+            if(b.getId() == bookId){
+                book = b;
+            }
+        }
+        return book;
+    }
+    public void addReader(String name){
+//        if(this.readers.isEmpty()){
+//            this.readers.add(new Reader(1, name));
+//        } else{
+//            this.readers.add(new Reader(this.readers.size()+1, name));
+//        }
+        this.registeredReaders.put(new Reader(this.registeredReaders.size()+1, name), new ArrayList<>());
+
+    }
+
+    public void printBookList(){
+        for(Book book : this.books){
+            System.out.println(book);
+        }
```

Comment:
Хороший кандидат на использование улучшенного `for loop`:
```suggestion
    books.forEach(System.out::println);
```